### PR TITLE
[admission-controller] Avoid scanner webhook configuration when scanner.enabled is set to false

### PR DESCRIPTION
**Pull request by dark-vex**
## Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [X] Chart Version bumped
- [ ] Variables are documented in the README.md
- [X] Title of the PR starts with chart name (e.g. `[mychartname]`)
- [X] PR only contains changes for one chart

When deploying the admission controller with just k8s audit detection, so with following flags:
```
--set features.k8sAuditDetections=true
--set scanner.enabled=false
```
Our template will configure anyway webhook configuration for `scanning.secure.sysdig.com` for ValidatingWebhookConfiguration object which causes to have logs full of these entries (this is an example from a k3s cluster):

```
Jun 06 11:47:32 prod-k3s-master k3s[402708]: E0606 11:47:32.241735  402708 dispatcher.go:149] failed calling webhook "scanning.secure.sysdig.com": failed to call webhook: the server could not find the requested resource
Jun 06 11:47:32 prod-k3s-master k3s[402708]: W0606 11:47:32.268591  402708 dispatcher.go:142] Failed calling webhook, failing open scanning.secure.sysdig.com: failed calling webhook "scanning.secure.sysdig.com": failed to call webhook: the server could not find the requested resource
Jun 06 11:47:32 prod-k3s-master k3s[402708]: E0606 11:47:32.269515  402708 dispatcher.go:149] failed calling webhook "scanning.secure.sysdig.com": failed to call webhook: the server could not find the requested resource
Jun 06 11:47:32 prod-k3s-master k3s[402708]: W0606 11:47:32.296006  402708 dispatcher.go:142] Failed calling webhook, failing open scanning.secure.sysdig.com: failed calling webhook "scanning.secure.sysdig.com": failed to call webhook: the server could not find the requested resource
Jun 06 11:47:32 prod-k3s-master k3s[402708]: E0606 11:47:32.296063  402708 dispatcher.go:149] failed calling webhook "scanning.secure.sysdig.com": failed to call webhook: the server could not find the requested resource
Jun 06 11:47:32 prod-k3s-master k3s[402708]: W0606 11:47:32.318974  402708 dispatcher.go:142] Failed calling webhook, failing open scanning.secure.sysdig.com: failed calling webhook "scanning.secure.sysdig.com": failed to call webhook: the server could not find the requested resource
Jun 06 11:47:32 prod-k3s-master k3s[402708]: E0606 11:47:32.319034  402708 dispatcher.go:149] failed calling webhook "scanning.secure.sysdig.com": failed to call webhook: the server could not find the requested resource
Jun 06 11:47:32 prod-k3s-master k3s[402708]: W0606 11:47:32.349677  402708 dispatcher.go:142] Failed calling webhook, failing open scanning.secure.sysdig.com: failed calling webhook "scanning.secure.sysdig.com": failed to call webhook: the server could not find the requested resource
Jun 06 11:47:32 prod-k3s-master k3s[402708]: E0606 11:47:32.350139  402708 dispatcher.go:149] failed calling webhook "scanning.secure.sysdig.com": failed to call webhook: the server could not find the requested resource
```

### CODE REVIEW

**Comment by dark-vex** [comment]
```
helm template admission-controller --set clusterName=k8s-daniele-vms --set features.k8sAuditDetections=true --set scanner.enabled=false --set sysdig.existingSecureAPITokenSecret=sysdig-admission-controller-webhook --set sysdig.url=https://<backend-url>

---
# Source: admission-controller/templates/webhook/admissionregistration.yaml
apiVersion: admissionregistration.k8s.io/v1
kind: ValidatingWebhookConfiguration
metadata:
  name: release-name-admission-controller-webhook
  annotations:
    "helm.sh/hook": "post-install, post-upgrade"
webhooks:
- name: audit.secure.sysdig.com
  matchPolicy: Equivalent
  rules:
  - apiGroups:
    - ""
    - apps
    - autoscaling
    - batch
    - networking.k8s.io
    - rbac.authorization.k8s.io
    - extensions
    apiVersions:
    - '*'
    operations:
    - '*'
    resources:
    - '*/*'
    scope: '*'
  clientConfig:
    service:
      namespace: default
      name: release-name-admission-controller-webhook
      path: /k8s-audit
    caBundle: LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSURWRENDQWp5Z0F3SUJBZ0lSQU15bkk5S2pnUXV3NXA1a0NFaVZWNHN3RFFZSktvWklodmNOQVFFTEJRQXcKTkRFeU1EQUdBMVVFQXhNcGNtVnNaV0Z6WlMxdVlXMWxMV0ZrYldsemMybHZiaTFqYjI1MGNtOXNiR1Z5TFhkbApZbWh2YjJzd0hoY05Nakl3TmpBMk1UQXpNVFU0V2hjTk16SXdOakF6TVRBek1UVTRXakEwTVRJd01BWURWUVFECkV5bHlaV3hsWVhObExXNWhiV1V0WVdSdGFYTnphVzl1TFdOdmJuUnliMnhzWlhJdGQyVmlhRzl2YXpDQ0FTSXcKRFFZSktvWklodmNOQVFFQkJRQURnZ0VQQURDQ0FRb0NnZ0VCQU9kUW8rNWlRRGZ1emdMS2V4UTBhQ1dLcEhMcApyeEFVbWdmaVE5TGRZc0NjLzY2Z0ljdndXODMweXhvVlh6RnlMZko3T0tzalZqdHFTdEJ1RFo3d2hhaDYxZkhTCi83SkFSd0hhcUdtdHlvcUc0ZnF0SjcxQUMzeC9OOGo2Z1NtREI0Rm1ranZNQkZGaFhjd0Urbk9UbE1SK0VLZUcKWXVrRkh5a3cwdGtoWFhXTUFYTE5ZZGs4Uko0czRLNElIVEJaODJzWUpZVFAyY2VuTUF1eGJGMGJsQ0NlNklPVgpaQXlhQUhnK0JXVWRzQ2k5Y3BsTjZSUHlzNXJWODNySGNuTk5rKytZZjUzRXYxSFF6SlZFcmgwcmZHcm03S29nCjBuUDQyZExETCtDeFBOcUptWkxqOXZXSlROck1kUWJZTlZ1NmtJamxzMnNiTlhWdmVzR0tPTWUvVkRjQ0F3RUEKQWFOaE1GOHdEZ1lEVlIwUEFRSC9CQVFEQWdLa01CMEdBMVVkSlFRV01CUUdDQ3NHQVFVRkJ3TUJCZ2dyQmdFRgpCUWNEQWpBUEJnTlZIUk1CQWY4RUJUQURBUUgvTUIwR0ExVWREZ1FXQkJRUDJndE15Q1ZEZzNVMktrbHdSWnNGClUrYU8xVEFOQmdrcWhraUc5dzBCQVFzRkFBT0NBUUVBR010WEJBSjd5SDVoeE0vSVliSFN0TDNRdy90UVY4c0kKaTJsMUVyeGhBMTJ4M2lWTEE5M3FtQnlCenRFU2JqTGY5ZXYxbmZra09lNUVCYkEzZytJdHpGMGk2bEd2NXpvTAp3VDBhcmhNZGh2dFlHVmFISDZ4eEN5MnlpTU9TbS9yWHRUQmVZUEVCTGgwQ3FrNnRrL2crcDZwVGc4S1RmN1dqCjFHZkVyM1JnajNrWjFQb3E3eG1DN1dCYXAvYlpsMDlvWFFycUxnQ1R2M3Z6UWFUcjgvNDFIcEtrMWFlTDB4VXIKaEJwR2didC84MUJtS0pKNVY3MGx1eDdreVJlSkYyb29CblV6NmRRaEUzWG8xWk41SVkxVWthWVdIcDk2UFJjYwpjUmJFSEtlQ1pwR0xLL2NWRkJTUHB6THMxSE83TmJGd0VTT3hTTmVUY3lrWUEwRHNyM0J2QXc9PQotLS0tLUVORCBDRVJUSUZJQ0FURS0tLS0tCg==
  admissionReviewVersions: ["v1", "v1beta1"]
  sideEffects: None
  timeoutSeconds: 5
  failurePolicy: Ignore
```

File: charts/admission-controller/templates/webhook/admissionregistration.yaml [hunk 1]
```diff
@@ -12,6 +12,7 @@ metadata:
   annotations:
     "helm.sh/hook": "post-install, post-upgrade"
 webhooks:
+{{- if .Values.scanner.enabled -}}
```
Comment (tembleking) [inline charts/admission-controller/templates/webhook/admissionregistration.yaml]: The linter seems to be failing, could you try to remove the last `-}}` and leave it as `}}`?

**Review by tembleking — CHANGES_REQUESTED** [review]

File: charts/admission-controller/templates/webhook/admissionregistration.yaml (hunk 1 above)
Comment (dark-vex) [reply charts/admission-controller/templates/webhook/admissionregistration.yaml]: @tembleking yes, absolutely, I did the suggested change :)

**Comment by tembleking** [comment]
Now it's all working, thanks @dark-vex!

**Pull request merged**